backend/src/middleware/middleware.js

```diff
@@ -0,0 +1,48 @@
+const bcrypt = require('bcrypt');
+
+const { User } = require("../Schema")
+
+module.exports = function(req, res, next) {
+    const newUser = new User(req.body)
+
+    const errFill = new Error('Please fill all the input')
+    const errSave = new Error('Failled data saving')
+    if (newUser.username === '') {
```

Comment:
You should avoid all these manual checks, by using the mongoose.validate() method. Check out this example over here:
https://mongoosejs.com/docs/validation.html

Validation is performed automatically by default whenever the User.save() method is called. But in this case we'd like to call it explicitly in our middleware. Let me know if this doesn't make sense.